metabase/metabase
# Old data picker in notebook for embedding

Demo https://www.loom.com/share/175d961bfc8d4bca8948bb5ea17c238d

Uses the `DataSelector` component for data source selection in the notebook editor for embedding cases - both full app embedding and the sdk. In other cases the existing `EntityPicker` is used without changes. 

**The old component doesn't support some newer features introduced in the `EntityPicker`:**
- Hiding specific data source types,
- Hiding data sources for different databases (used for joins).

There are e2e tests for all interesting data selection cases I'm aware of. If there is some case missing please let me know.

How to verify:
- Run EE
- Enable interactive embedding in admin settings. Use `http://localhost:3001` for the allowed iframe host.
- Add a html page `iframe.html` with the following contents
```html
<!DOCTYPE html>
<html>
<head>
  <title>Full app embedding</title>
</head>
<body>
<iframe src="http://localhost:3000?new_button=true" width="1000" height="800"></iframe>
</body>
</html>
```
- Run some simple http server like `npx http-server -p 3001` and open `http://localhost:3001/iframe.html`
- Make sure that you can pick a data source in the notebook editor using the old picker
- Make sure that the main app itself is not affected 

<img width="1032" alt="Screenshot 2024-11-28 at 10 42 17" src="https://github.com/user-attachments/assets/c98796a0-a959-4228-8ef5-72d3528e5db5">
<img width="1100" alt="Screenshot 2024-11-28 at 10 42 28" src="https://github.com/user-attachments/assets/26926cb0-ece9-4925-9a24-d8c8b1370e41">
